consensus/src/chained_bft/network.rs

```diff
@@ -283,11 +286,26 @@ impl<T: Payload> NetworkTask<T> {
         self_receiver: channel::Receiver<failure::Result<Event<ConsensusMsg>>>,
         validators: Arc<ValidatorVerifier>,
     ) -> (NetworkTask<T>, NetworkReceivers<T>) {
-        let (proposal_tx, proposal_rx) = channel::new(1_024, &counters::PENDING_PROPOSAL);
-        let (vote_tx, vote_rx) = channel::new(1_024, &counters::PENDING_VOTES);
-        let (block_request_tx, block_request_rx) =
-            channel::new(1_024, &counters::PENDING_BLOCK_REQUESTS);
-        let (sync_info_tx, sync_info_rx) = channel::new(1_024, &counters::PENDING_SYNC_INFO_MSGS);
+        let (proposal_tx, proposal_rx) = libra_channel::new(PerValidatorQueue::new(
+            QueueStyle::LIFO,
+            1,
+            Some(&counters::PROPOSAL_DROPPED_MSGS),
+        ));
+        let (vote_tx, vote_rx) = libra_channel::new(PerValidatorQueue::new(
+            QueueStyle::LIFO,
+            1,
+            Some(&counters::VOTES_DROPPED_MSGS),
+        ));
+        let (block_request_tx, block_request_rx) = libra_channel::new(PerValidatorQueue::new(
+            QueueStyle::LIFO,
+            1,
```

Comment:
I wonder if a max queue size of 1 is a little big aggressive. are we sure there are no cases where someone might send us, e.g., multiple concurrent block retrieval requests? there are also few freshness constraints on a block retrieval request, apart from the timeout...